File: examples/companion-packages/java-maven-app/src/main/java/ai/agentxm/examples/pawmatch/Charities.java
package ai.agentxm.examples.pawmatch;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Locale;
import java.util.Map;
import java.util.Optional;

/** Static catalogue of animal welfare charities. */
public final class Charities {

    /** A charity record. */
    public record Charity(
            String slug,
            String name,
            String focus,
            String description,
            String url,
            String ratingNote) {}

    public static final List<Charity> ALL = List.of(
            new Charity(
                    "best-friends",
                    "Best Friends Animal Society",
                    "shelters",
                    "No-kill movement; supports adoptions, shelters, and advocacy nationwide.",
                    "https://bestfriends.org/donate",
                    "Charity Navigator 4-star"),
            new Charity(
                    "petsmart-charities",
                    "PetSmart Charities",
                    "shelters",
                    "Grants to local shelters; spay/neuter; adoption events.",
                    "https://petsmartcharities.org/donate",
                    "Charity Navigator 4-star (96% program ratio)"),
            new Charity(
                    "brother-wolf",
                    "Brother Wolf Animal Rescue",
                    "rescue",
                    "Local rescue with national-impact outreach programs.",
                    "https://bwar.org/donate",
                    "Charity Navigator 4-star, GuideStar Platinum"),
            new Charity(
                    "animal-welfare-institute",
                    "Animal Welfare Institute",
                    "policy",
                    "Policy and advocacy reducing cruelty inflicted on animals.",
                    "https://awionline.org/donate",
                    "Charity Navigator 4-star"),
            new Charity(
                    "aspca",
                    "ASPCA",
                    "shelters",
                    "Adoption, anti-cruelty programs, and animal welfare advocacy.",
                    "https://www.aspca.org/donate",
                    "Charity Navigator 4-star"));

    public static final String DISCLAIMER =
            "Curated example list — verify current ratings on Charity Navigator or GuideStar before giving.";

    private static final Map<String, Charity> BY_SLUG;

    static {
        Map<String, Charity> bySlug = new LinkedHashMap<>();
        for (Charity charity : ALL) {
            bySlug.put(charity.slug(), charity);
        }
        BY_SLUG = Map.copyOf(bySlug);
    }

    private Charities() {}

    public static Optional<Charity> findBySlug(String slug) {
        return Optional.ofNullable(BY_SLUG.get(slug.toLowerCase(Locale.ROOT)));
    }

    public static List<Charity> filterByFocus(String focus) {
        String target = focus.toLowerCase(Locale.ROOT);
        if (target.equals("all")) {
            return ALL;
        }
        return ALL.stream()
                .filter(c -> c.focus().toLowerCase(Locale.ROOT).equals(target))
                .toList();
    }
}

File: examples/companion-packages/maven-scala-app/src/main/scala/ai/agentxm/examples/pawmatch/Charities.scala
package ai.agentxm.examples.pawmatch

/** A charity record. */
final case class Charity(
    slug: String,
    name: String,
    focus: String,
    description: String,
    url: String,
    ratingNote: String,
)

/** Static catalogue of animal welfare charities. */
object Charities:
  val all: List[Charity] = List(
    Charity(
      slug = "best-friends",
      name = "Best Friends Animal Society",
      focus = "shelters",
      description =
        "No-kill movement; supports adoptions, shelters, and advocacy nationwide.",
      url = "https://bestfriends.org/donate",
      ratingNote = "Charity Navigator 4-star",
    ),
    Charity(
      slug = "petsmart-charities",
      name = "PetSmart Charities",
      focus = "shelters",
      description = "Grants to local shelters; spay/neuter; adoption events.",
      url = "https://petsmartcharities.org/donate",
      ratingNote = "Charity Navigator 4-star (96% program ratio)",
    ),
    Charity(
      slug = "brother-wolf",
      name = "Brother Wolf Animal Rescue",
      focus = "rescue",
      description = "Local rescue with national-impact outreach programs.",
      url = "https://bwar.org/donate",
      ratingNote = "Charity Navigator 4-star, GuideStar Platinum",
    ),
    Charity(
      slug = "animal-welfare-institute",
      name = "Animal Welfare Institute",
      focus = "policy",
      description = "Policy and advocacy reducing cruelty inflicted on animals.",
      url = "https://awionline.org/donate",
      ratingNote = "Charity Navigator 4-star",
    ),
    Charity(
      slug = "aspca",
      name = "ASPCA",
      focus = "shelters",
      description = "Adoption, anti-cruelty programs, and animal welfare advocacy.",
      url = "https://www.aspca.org/donate",
      ratingNote = "Charity Navigator 4-star",
    ),
  )

  val Disclaimer: String =
    "Curated example list — verify current ratings on Charity Navigator or GuideStar before giving."

  private val bySlug: Map[String, Charity] = all.map(c => c.slug -> c).toMap

  def findBySlug(slug: String): Option[Charity] = bySlug.get(slug.toLowerCase)

  def filterByFocus(focus: String): List[Charity] =
    if focus.equalsIgnoreCase("all") then all
    else
      val target = focus.toLowerCase
      all.filter(_.focus.toLowerCase == target)
